enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,397 @@
+   `openshift-config` namespace as the `etc-pki-entitlement` `Secret`.
+4. Run a pod with the `"subscription.openshift.io/cluster-bundle": rhel8` annotation, and a UBI8
+   container which performs a `yum install` of a RHEL package that requires a subscription.
+
+### Graduation Criteria
+
+This feature will likely require Dev Preview, Tech Preview, and GA maturity levels.
+
+#### Dev Preview
+
+- Cluster-level APIs at `v1alpha1`
+- Operator configuration APIs at `v1alpha1`
+- Dev preview release of the projected resource CSI driver via OLM (as community operator)
+- Release as a community OLM operator
+
+##### Dev Preview -> Tech Preview
+
+- Projected resource CSI driver reaches tech preview state
+- Cluster-level API reaches `v1beta1`
+- Clear documentation on how to enable subscription injection across the cluster
+- Initial scale testing
+- Gather feedback for namespace-restricted subscription bundles
+- Release as a tech preview, official OLM operator
+
+##### Tech Preview -> GA
+
+- All APIs reach `v1` stability
+- Projected resource CSI driver reaches GA
+- Security audit for these components and the projected resource CSI driver
+- Scale testing (with potential mitigations like `HorizontalPodAutoscalers`)
+- Disruption testing (with potential mitigations like `PodDisruptionBudgets`)
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled via OLM. Each released version will specify:
+
+- Minimum supported version of the projected resource CSI driver
+- Minimum supported kube version of the cluster.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components? What are the guarantees? Make sure
+this is in the test plan.
+
+Consider the following in developing a version skew strategy for this enhancement:
+
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and in the kubelet? How
+  does an n-2 kubelet without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI or CNI may require
+  updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Drawbacks
+
+- Since this is being deliverd via OLM, customers will need to opt into capabilities that they may
+  expect by default.
+- Users will have to specify on a per-workload basis which subscriptions should be added, increasing
+  toil.
+- Cluster admins need to manually obtain the entitlement keys and `redhat.repo` configurations.
+
+## Alternatives
+
+### Document what we have
+
+Today, any UBI/RHEL-based container can access subscription content if the `redhat.repo` file is
+mounted into `/run/secrets`, and the entitlement keys are mounted into
```

Comment:
Nothing requires the redhat.repo file in `/run/secrets` today.  Only the entitlement keys are required. The auto-mounted rhsm config is also honored inside the container, in case of a Satellite server instead of RH CDN.